### ISSUE TITLE
Ephemeral devices changing IPs don't work well with TS_TAILNET_TARGET_FQDN

### ISSUE THREAD

**Issue by gjhenrique** [tailscale/tailscale#11281]
### What is the issue?

We're experimenting with development environments where Tailscale will provide ingress/egress between the EKS cluster and the local services where developers will work on.

For exposing the local service, we're using the `k8s` operator [egress feature(https://tailscale.com/kb/1236/kubernetes-operator#expose-a-tailnet-node-to-your-cluster-using-its-tailscale-magicdns-name) with `tailscale.com/tailnet-fqdn` pointing to the MagicDNS name of the local container.

One important detail is that the local development containers will keep changing its IP because users might need to recreate their environment.

After the environment is rebuilt, we realized that the proxy doesn't redirect the incoming requests to the container anymore. This is due to the operator not cleaning up `nftables` properly when that device receives a new IP

For example:
```
> nft -a list ruleset

table ip nat { # handle 4
        ...
        chain PREROUTING { # handle 5
                type nat hook prerouting priority dstnat; policy accept;
                oifname != "tailscale0*" dnat to 100.126.36.111 # handle 8 (This is the first IP)
                oifname != "tailscale0*" dnat to 100.88.67.112 # handle 10 (This is the second IP)
        }
}

```

As soon as I delete the first rule with `nft delete rule nat PREROUTING handle 8`, things start working again.

I suspect using `iptables` won't trigger the bug because of ordering rules, but I didn't have time to test that assuption yet.

### Steps to reproduce

1. Start a local container with tailscale and set `TS_HOSTNAME` to always have the same name
1. Create a `kubernetes` service with `tailscale.com/tailnet-fqdn` pointing to a local container using tailscale
1. Verify from a pod that the cluster service is reachable (`curl <svc>.<namespace>.cluster.svc.local`)
1. Stop the local container and start a new one. This container will keep the same name, but receive a new IP
1. The traffic doesn't reach the container anymore due to nftables not cleaning up the first IP

### Are there any recent changes that introduced the issue?

I don't think so

### OS

Linux

### OS version

The proxy runs Alpine. Locally, we use containers with Ubuntu 

### Tailscale version

1.58.2

### Bug report

BUG-78410f7eada98e5ccc65f29e5efb5094bd331ff67698e056542186b43473b737-20240228124217Z-0263349d7b05e7a7

**Comment by irbekrm** [comment]
Thanks for reporting the issue.

> I suspect using iptables won't trigger the bug because of ordering rules, but I didn't have time to test that assuption yet.

You're right that iptables shouldn't have that problem as with iptables we [explicitly insert the new rule at first position](https://github.com/tailscale/tailscale/blob/v1.60.0/util/linuxfw/iptables_runner.go#L305-L308).
We should do a similar thing with [nftables](https://github.com/tailscale/tailscale/blob/v1.60.0/util/linuxfw/nftables_runner.go#L143-L178), perhaps insert the rule at a specific handle, to ensure if the destination address changes, the rule is replaced rather than a new one appended.

If you want to use the iptables whilst we fix this, you can do so by setting `.values.proxyConfig.firewallMode` to `iptables` (if using helm) or `PROXY_FIREWALL_MODE` for static operator yaml.

**Comment by gjhenrique** [comment]
Hi @irbekrm

Your configuration worked. We will use `iptables` for now.

Thanks

**Comment by irbekrm** [comment]
@gjhenrique this turned out to be an easy (hopefully) fix in #11303 

`1.61.67` Tailscale operator helm chart/`unstable-v1.61.67` image tags have the fix- do let us know if you see further issues?

**Comment by irbekrm** [comment]
Should be fixed in #11303
